includes/template-functions.php

```diff
@@ -1139,3 +1139,47 @@ function edd_download_shortcode_item( $atts, $i ) {
 	edd_get_template_part( 'shortcode', 'download' );
 }
 add_action( 'edd_download_shortcode_item', 'edd_download_shortcode_item', 10, 2 );
+
+/**
+ * Load the pagination for the [downloads] shortcode.
+ *
+ * @since 2.9.7
+ *
+ * @param array  $atts The [downloads] shortcode attributes.
+ * @param object $downloads The WP_Query.
+ * @param array  $query EDD's array of attributes used to construct the main WP_Query.
+ */
+function edd_downloads_pagination( $atts, $downloads, $query ) {
+
+	if ( filter_var( $atts['pagination'], FILTER_VALIDATE_BOOLEAN ) ) :
+
+		$pagination = false;
+
+		if ( is_single() ) {
+			$pagination = paginate_links( apply_filters( 'edd_download_pagination_args', array(
+				'base'    => get_permalink() . '%#%',
+				'format'  => '?paged=%#%',
+				'current' => max( 1, $query['paged'] ),
+				'total'   => $downloads->max_num_pages
+			), $atts, $downloads, $query ) );
+		} else {
+			$big = 999999;
+			$search_for   = array( $big, '#038;' );
+			$replace_with = array( '%#%', '&' );
+			$pagination = paginate_links( apply_filters( 'edd_download_pagination_args', array(
+				'base'    => str_replace( $search_for, $replace_with, get_pagenum_link( $big ) ),
+				'format'  => '?paged=%#%',
+				'current' => max( 1, $query['paged'] ),
+				'total'   => $downloads->max_num_pages
+			), $atts, $downloads, $query ) );
+		}
+
+		if ( ! empty( $pagination ) ) : ?>
+		<div id="edd_download_pagination" class="navigation">
```

Comment:
The ID `edd_download_pagination` isn't new with this PR. It's already being used in core, no change. So this is one we'll have to keep as-is.

Reply:
👍 Completely agree on 1 function to rule them all, just didn't see it as part of this issue's objective (more of an after thing).

Reply:
@amdrew I would argue that the issue is perfectly in the scope to break it out. The title of the PR itself is "Decouple pagination from downloads shortcode".

That signifies to me we should do this now, instead of later, as it means we're setting a pattern for pagination generation, and providing an example at the same time, in the refactor.

Further the issue itself states "Reposition/refactor the pagination...", so this seems like a perfect time to break this out. Breaking it out would also allow us to make this testable for how many pages it should have.

Reply:
@cklosowski 👍I'll work on the pagination further and aim to get all pagination running through it.